uint16 msg_seq
int16 location